src/core/broker_config.cpp

```diff
@@ -17,80 +17,282 @@
+    [](auto& config) { config.settings.disc_attributes.dns_model = "RDMnet Broker Service"; }
+  },
+  {
+    "/dns_sd/scope"_json_pointer,
+    json::value_t::string,
+    [](const json& val, auto& config) { return ValidateAndStoreString(val, config.settings.disc_attributes.scope, E133_SCOPE_STRING_PADDED_LENGTH -1, false); },
+    std::function<void(BrokerConfig&)>() // Leave the default constructed scope value.
+  },
+  {
+    "/max_connections"_json_pointer,
+    json::value_t::number_unsigned,
+    [](const json& val, auto& config) { return ValidateAndStoreInt<unsigned int>(val, config.settings.max_connections); },
+    std::function<void(BrokerConfig&)>() // Leave the default constructed value in the settings struct
+  },
+  {
+    "/max_controllers"_json_pointer,
+    json::value_t::number_unsigned,
+    [](const json& val, auto& config) { return ValidateAndStoreInt<unsigned int>(val, config.settings.max_controllers); },
+    std::function<void(BrokerConfig&)>() // Leave the default constructed value in the settings struct
+  },
+  {
+    "/max_controller_messages"_json_pointer,
+    json::value_t::number_unsigned,
+    [](const json& val, auto& config) { return ValidateAndStoreInt<unsigned int>(val, config.settings.max_controller_messages); },
+    std::function<void(BrokerConfig&)>() // Leave the default constructed value in the settings struct
+  },
+  {
+    "/max_devices"_json_pointer,
+    json::value_t::number_unsigned,
+    [](const json& val, auto& config) { return ValidateAndStoreInt<unsigned int>(val, config.settings.max_devices); },
+    std::function<void(BrokerConfig&)>() // Leave the default constructed value in the settings struct
+  },
+  {
+    "/max_device_messages"_json_pointer,
+    json::value_t::number_unsigned,
+    [](const json& val, auto& config) { return ValidateAndStoreInt<unsigned int>(val, config.settings.max_device_messages); },
+    std::function<void(BrokerConfig&)>() // Leave the default constructed value in the settings struct
+  },
+  {
+    "/max_reject_connections"_json_pointer,
+    json::value_t::number_unsigned,
+    [](const json& val, auto& config) { return ValidateAndStoreInt<unsigned int>(val, config.settings.max_reject_connections); },
+    std::function<void(BrokerConfig&)>() // Leave the default constructed value in the settings struct
+  }
+};
+// clang-format on
+
+// Read the JSON configuration from a file.
+BrokerConfig::ParseResult BrokerConfig::Read(const std::string& file_name)
 {
   std::ifstream file_stream(file_name);
   if (!file_stream.is_open())
     return ParseResult::kFileOpenErr;
 
-  return ReadFromStream(file_stream);
+  return Read(file_stream);
 }
 
-BrokerConfig::ParseResult BrokerConfig::ReadFromStream(std::istream& stream)
+// Read the JSON configuration from an input stream.
+BrokerConfig::ParseResult BrokerConfig::Read(std::istream& stream)
 {
   try
```

Comment:
I don't think this try is necessary anymore now that you are catching json::parse_errors inside of the ValidateCurrent function.